com.reprezen.swagedit/src/com/reprezen/swagedit/model/NodeDeserializer.java

```diff
@@ -0,0 +1,84 @@
+package com.reprezen.swagedit.model;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+public class NodeDeserializer extends JsonDeserializer<AbstractNode> {
+
+    @Override
+    public AbstractNode deserialize(JsonParser p, DeserializationContext context)
+            throws IOException, JsonProcessingException {
+
+        final Model model = (Model) context.getAttribute("model");
+        final AbstractNode parent = (AbstractNode) context.getAttribute("parent");
```

Comment:
The magic strings `parent` and `pointer` are used at least three times. What about converting them into reusable constants?